Lib/test/test_codecs.py

```diff
@@ -2754,29 +2754,18 @@ def test_uu_invalid(self):
 
 def _get_test_codec(codec_name):
     return _TEST_CODECS.get(codec_name)
-codecs.register(_get_test_codec) # Returns None, not usable as a decorator
-
-try:
-    # Issue #22166: Also need to clear the internal cache in CPython
-    from _codecs import _forget_codec
-except ImportError:
-    def _forget_codec(codec_name):
-        pass
 
 
 class ExceptionChainingTest(unittest.TestCase):
 
     def setUp(self):
-        # There's no way to unregister a codec search function, so we just
-        # ensure we render this one fairly harmless after the test
-        # case finishes by using the test case repr as the codec name
-        # The codecs module normalizes codec names, although this doesn't
-        # appear to be formally documented...
-        # We also make sure we use a truly unique id for the custom codec
-        # to avoid issues with the codec cache when running these tests
-        # multiple times (e.g. when hunting for refleaks)
-        unique_id = repr(self) + str(id(self))
-        self.codec_name = encodings.normalize_encoding(unique_id).lower()
+        # After the codecs.unregister function is added in bpo-41842,
+        # we don't make sure we use a truly unique id for the custom codec
+        # to avoid the refleaks of the codec cache when running these tests
+        # multiple times.
+        self.codec_name = encodings.normalize_encoding(repr(self).lower())
```

Comment:
We don't need any comment anymore, not the complex code to generate a complex unique name:

```suggestion
        self.codec_name = 'exception_chaining_test'
```

Reply:
Copy that, updated.